src/ansi.rs

```diff
@@ -298,108 +270,45 @@ impl<T: Timeout> Processor<T> {
 
     /// Synchronized update timeout.
     pub fn sync_timeout(&self) -> &T {
-        &self.state.sync_state.timeout
+        &self.state.sync_timeout
     }
 
     /// Process a new byte from the PTY.
     #[inline]
-    pub fn advance<H>(&mut self, handler: &mut H, byte: u8)
+    pub fn advance<H>(&mut self, handler: &mut H, bytes: &[u8])
     where
         H: Handler,
     {
-        if self.state.sync_state.timeout.pending_timeout() {
-            self.advance_sync(handler, byte);
-        } else {
-            let mut performer = Performer::new(&mut self.state, handler);
-            self.parser.advance(&mut performer, byte);
-        }
-    }
+        let mut performer = Performer::new(&mut self.state, handler, &mut self.sync_handler);
+        self.parser.advance(&mut performer, bytes);
 
-    /// End a synchronized update.
-    pub fn stop_sync<H>(&mut self, handler: &mut H)
-    where
-        H: Handler,
-    {
-        // Process all synchronized bytes.
-        for i in 0..self.state.sync_state.buffer.len() {
-            let byte = self.state.sync_state.buffer[i];
-            let mut performer = Performer::new(&mut self.state, handler);
-            self.parser.advance(&mut performer, byte);
+        // TODO: Is this great? :/
```

Comment:
Not entirely sure about this yet. We check before and after every `advance` (here and in `Performer::new`) whether we need to restart/terminate synchronized updates. This seems kinda inefficient but I won't look into it before we have some general performance numbers available.